karton/core/backend/direct.py

```diff
@@ -33,85 +33,31 @@
 KARTON_TASK_NAMESPACE = "karton.task"
 KARTON_OUTPUTS_NAMESPACE = "karton.outputs"
 
-KartonBind = namedtuple(
-    "KartonBind",
-    [
-        "identity",
-        "info",
-        "version",
-        "persistent",
-        "filters",
-        "service_version",
-        "is_async",
-    ],
-)
-
-
 KartonOutputs = namedtuple("KartonOutputs", ["identity", "outputs"])
 logger = logging.getLogger(__name__)
 
 
-class KartonMetrics(enum.Enum):
-    TASK_PRODUCED = "karton.metrics.produced"
-    TASK_CONSUMED = "karton.metrics.consumed"
-    TASK_CRASHED = "karton.metrics.crashed"
-    TASK_ASSIGNED = "karton.metrics.assigned"
-    TASK_GARBAGE_COLLECTED = "karton.metrics.garbage-collected"
-
-
-@dataclasses.dataclass(frozen=True, order=True)
-class KartonServiceInfo:
-    """
-    Extended Karton service information.
-
-    Instances of this dataclass are meant to be aggregated to count service replicas
-    in Karton Dashboard. They're considered equal if identity and versions strings
-    are the same.
-    """
-
-    identity: str = dataclasses.field(metadata={"serializable": False})
-    karton_version: str
-    service_version: Optional[str] = None
-    # Extra information about Redis client
-    redis_client_info: Optional[Dict[str, str]] = dataclasses.field(
-        default=None, hash=False, compare=False, metadata={"serializable": False}
+def make_redis_client_name(service_info: KartonServiceInfo) -> str:
+    params = {
```

Comment:
Is this validated somewhere else? Parsing the name won't work when identity has `?`:

```suggestion
    assert '?' not in service_info.identity, "Identity can't contain `?` character"
    params = {
```

Reply:
Yes, this is checked by `KartonBackendBase._validate_identity`, so we can assume that identity doesn't contain that character